wherehows-web/app/components/dataset-aclaccess.js

```diff
@@ -0,0 +1,138 @@
+import Component from '@ember/component';
+import { computed, getProperties, setProperties, get, set } from '@ember/object';
+import { requestAclAccess } from 'wherehows-web/utils/api/datasets/acl-access';
+import { accessState, pickList } from 'wherehows-web/constants/dataset-aclaccess';
+
+export default Component.extend({
+  /**
+   * Define property to bind textarea
+   * @type {string}
+   */
+  requestReason: '',
+
+  /**
+   * Define the property for the initial page  
+   * @type {object} 
+   */
+  accessInfo: null,
+
+  /**
+   * Define the property for reloading the page when a user applies for permission.
+   * @type {object} 
+   */
+  accessResponse: null,
+
+  /**
+   * Define the property to identify user 
+   * @type {string} 
+   */
+  currentUser: null,
+
+  /**
+   * Define the property for loading spinner component
+   * @type {boolean} 
+   */
+  isLoadSpinner: false,
```

Comment:
Rather than using this flag to manually track the state of the async operation, let's use the installed package [ember-concurrency](https://github.com/linkedin/WhereHows/blob/master/wherehows-web/package.json#L57) for these types of tasks. Reduces the boilerplate and opportunities for mistakes